test/Libraries/RevitIntegrationTests/RegressionTests.cs

```diff
@@ -79,6 +82,155 @@ public void Regressions(RegressionTestData testData)
             }
         }
 
+        [Test]
+        [TestModel(@".\empty.rfa")]
+        public void NoUINodesLeftBehind()
+        {
+            // This verifies Revit UI Nodes are not lost in translation by comparing
+            // the overall node count on the active ws pre and post save
+
+            // Assertion variables
+            int preSaveNodeCount = 0;
+            int postSaveNodeCount = 0;
+
+            // Setup home ws
+            var homespace = Model.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.NotNull(homespace, "The current workspace is not a HomeWorkspaceModel");
+
+            // Iterate through all loaded nodes in library & add Revit UI nodes to ws
+            var nodeList = Model.SearchModel.SearchEntries.OfType<NodeSearchElement>();
+            foreach (var node in nodeList)
+            {
+                var assembly = Path.GetFileName(node.Assembly);
+
+                var searchElement = node as NodeSearchElement;
+                if (assembly == "DSRevitNodesUI.dll")
```

Comment:
all of the nodes touched in this PR live in 'DSRevitNodesUI.dll` but these tests should still probably run for all D4R nodes.  It may be helpful to eventually add something similar in Core as well to catch these issues when nodes are added/modified

Reply:
did you get it to load nodes from all dlls yet?

Reply:
not yet, I am still a bit unsure what assemblies are being loaded into the library during the tests.  I need to debug further but didn't want to hold up this PR since the test is covering the changes I made.  I will look into it further so we can apply this in other areas